components/tools/OmeroJava/test/integration/AbstractServerTest.java

```diff
@@ -285,6 +298,175 @@ public void tearDown() throws Exception {
     }
 
     /**
+     * Cast the map of strings (e.g. {@link #ALL_GROUPS_CONTEXT})
+     * into implicit context, which asks for {@code <String, String>}.
+     * @param implicitContext the implicit context to be changed
+     * @param newContext the map of strings (e.g. {@link #ALL_GROUPS_CONTEXT})
+     */
+    protected void mergeIntoContext(ImplicitContext implicitContext, Map<String, String> newContext) {
+        for (final Entry<String, String> entry : newContext.entrySet()) {
+            implicitContext.put(entry.getKey(), entry.getValue());
+        }
+    }
+
+    /**
+     * An enumeration of properties for which IObjects can be examined.
+     * Used in {@link AbstractServerTest.verifyObjectProperty}.
+     * @author pwalczysko@dundee.ac.uk
+     */
+    private enum Properties {
```

Comment:
Usually enums are named for the singular; maybe refactor it to be called `DetailsProperty`?